packages/api/src/modules/providers/demarchesSimplifiees/adapters/DemarchesSimplifieesEntityAdapter.ts

```diff
@@ -0,0 +1,44 @@
+import * as lodash from "lodash";
+import { DemandeSubvention, ProviderValue } from "@api-subventions-asso/dto";
+import { isString } from "lodash";
+import ProviderValueFactory from "../../../../shared/ProviderValueFactory";
+import demarchesSimplifieesService from "../demarchesSimplifiees.service";
+import DemarchesSimplifieesDataEntity from "../entities/DemarchesSimplifieesDataEntity";
+import DemarchesSimplifieesMapperEntity from "../entities/DemarchesSimplifieesMapperEntity";
+import { isValidDate } from "../../../../shared/helpers/DateHelper";
+import { DefaultObject } from "../../../../@types";
+
+export class DemarchesSimplifieesEntityAdapter {
+    static toSubvention(
+        entity: DemarchesSimplifieesDataEntity,
+        mapper: DemarchesSimplifieesMapperEntity
+    ): DemandeSubvention {
+        // Used for decembre month (décembre to decembre)
+        const cleanDate = dateString => new Date(dateString.replace("é", "e"));
+        const toPv = ProviderValueFactory.buildProviderValueAdapter(
+            demarchesSimplifieesService.provider.name,
+            new Date(entity.demande.dateDerniereModification)
+        );
+
+        const subvention: DefaultObject<ProviderValue> = {
+            siret: toPv(entity.siret)
+        };
+
+        mapper.schema.forEach(property => {
+            let value = lodash.get(entity, property.from);
+            const valueDate = new Date(cleanDate(value));
+            const valueNumber = parseFloat(value);
+
+            if (value === undefined || value === "") return;
+            else if (isValidDate(valueDate)) value = valueDate;
+            else if (!isNaN(valueNumber)) value = valueNumber;
+
+            lodash.set(subvention, property.to, toPv(value));
```

Comment:
idem pourquoi pas `subvention[property.to] = toPv(value)` ?

Reply:
Même reponse quand haut